src/app/pages/marker-generator/marker-generator.component.ts

```diff
@@ -0,0 +1,222 @@
+
+const blankSVG = '<?xml version="1.0" encoding="UTF-8"?><svg xmlns="http://www.w3.org/2000/svg" width="1" height="1"/>';
+
+const MARKERS_VERSION = '4';
+
+const BARCODE_TYPE = MATRIX_4X4_BCH_1393;
+
+interface Page {
+  value: string;
+  type: string;
+  backgroundSVG: string;
+}
+
+interface OrientationAndNumPerPage {
+  pageOrientation: PageOrientation;
+  numPerPage: 1 | 2;
+}
+
+@Component({
+  selector: 'app-marker-generator',
+  templateUrl: './marker-generator.component.html',
+  styleUrls: ['./marker-generator.component.scss']
+})
+export class MarkerGeneratorComponent implements OnInit {
+
+  maxNumNests = MAX_NEST_MARKER - MIN_NEST_MARKER + 1;
+
+
+  markerFormGroup = new FormGroup({
+    pageSizeControl: new FormControl('LETTER', Validators.required),
+    pageOrientationControl: new FormControl({ pageOrientation: 'portrait', numPerPage: 1 }, Validators.required),
+    includeFlowersControl: new FormControl(true, Validators.required),
+    includeNestsControl: new FormControl(true, Validators.required),
+    numNestsControl: new FormControl(30, [
+      Validators.required,
+      Validators.min(1),
+      Validators.max(this.maxNumNests)
+    ]),
+  },
+    {
+      validators: (fg: FormGroup) =>
+        (fg.controls.includeFlowersControl.value || fg.controls.includeNestsControl.value) ? null : { 'no-flowers-or-nests': true }
+    });
+
+  compareOrientationAndNumPerPage(a: OrientationAndNumPerPage, b: OrientationAndNumPerPage) {
+    return a.pageOrientation === b.pageOrientation && a.numPerPage === b.numPerPage;
+  }
+
+  constructor(public sanitizer: DomSanitizer) { }
+
+  ngOnInit(): void {
+
+  }
+
+
+  getBarcode(value: number) {
+    return new CustomBarcodeMarkerGenerator(BARCODE_TYPE, value);
+  }
+
+  flowerPages(markerSize: number) {
+    return rangeArray(MIN_FLOWER_MARKER, MAX_FLOWER_MARKER).map(val =>
+    ({
+      value: `${val}`,
+      type: 'Flower',
+      backgroundSVG: this.getBarcode(val).asSVGWithSize(markerSize)
+    }));
+  }
+
+  nestPages(nestCount: number, markerSize: number) {
+    if (nestCount === 0) {
+      return [];
+    }
+    return rangeArray(MIN_NEST_MARKER, MIN_NEST_MARKER + nestCount - 1).map(val =>
+    ({
```

Comment:
```suggestion
    return rangeArray(MIN_NEST_MARKER, MIN_NEST_MARKER + nestCount - 1).map(val => ({
```
Ditto. Incidentally, I like your use of `rangeArray()` here!